Configure FHIR Version

jafeltra opened the pull request:
This PR supports selecting the FHIR version in the Configuration modal for SUSHI to use. Only R4 and R4B are supported because they are the only official FHIR versions that are on `packages.fhir.org` at the moment. Eventually R5 will make it there, and it can be supported in FSH Online at that time.

The configuration for FHIR version should get shared and reloaded when using a "Share FSH" link.

cmoesel reviewed (approved):
This looks good to me! I like the way it looks in the config and the implementation seems to work great. I tested using `Convert to JSON` and `Convert to FSH` -- and also using share URLs (old and new). It all looks good to me!

jafeltra commented on src/components/FSHControls.js:
```diff
@@ -329,12 +350,25 @@ export default function FSHControls(props) {
             fullWidth
             label="Dependencies"
-            helperText="Format: packageId#version, packageId#version (e.g., hl7.fhir.us.core#3.1.1)"
+            helperText="Format: packageId#version, packageId#version (e.g., hl7.fhir.us.core#5.0.1)"
```
Same! So I had to seize the moment!

mint-thompson reviewed (approved):
Tested it out locally, and everything looks good and works correctly. Thank you!

jafeltra commented:
@cmoesel  and @mint-thompson - I found and fixed a bug in this branch, so I think this would benefit from a re-review. The issue was that I forgot to include a default FHIR version to use and load when running GoFSH. I noticed this bug when I was converting one instance to FSH. No FHIR definitions were loaded, so the resource it was an instance of could not be found, and no FSH rules were being created. I had moved the place where we choose a FHIR version to load out of the functions called in `loadAndCleanDatabase` and into the `runSUSHI` and (now) `runGoFSH` so we can properly load the configured version.

In doing this, I also realized that the dependencies we passed into GoFSH weren't formatted properly, but things were working fine because of how we re-parse things later on. I ended up fixing it now because I saw it was wrong and fixing it made the logic for adding the configured FHIR version easier.

One other thing I wanted to call out after looking at this more - when you run GoFSH, it'll use your configured FHIR version or figure out a default version. However, once it does, it'll load that FHIR version into the configuration modal, even if you didn't explicitly pick a version. I remember thinking this was okay when I originally implemented this, but wanted to make sure you both agreed.

jafeltra commented:
Also, I added a `.tools-version` file. I think we should update the Node version this uses eventually, but I figured I'd at least include what we're using for now.

cmoesel commented:
> The issue was that I forgot to include a default FHIR version to use and load when running GoFSH.

I swear I had tested this and it worked! Maybe the FHIR definitions were already loaded and I didn't fully refresh or something... But I _did_ think of this; I just apparently was really bad at actually testing it!

> it'll load that FHIR version into the configuration modal, even if you didn't explicitly pick a version.

Yeah, I think that's probably good -- because if it's R5, you should be able to convert to FSH and then convert to JSON (roundtrip). If it didn't remember the FHIR version in the config, then the convert to JSON might use the wrong version.

cmoesel reviewed (approved):
Watch out! Super cool feature on its way to production!
